include/openenclave/bits/sgx/region.h

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_INTERNAL_REGION_H
+#define _OE_INTERNAL_REGION_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/sgx/sgxtypes.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+#define OE_MAX_REGIONS 16
+
+typedef struct oe_region_context oe_region_context_t;
+
+/**
+ * Adds new memory regions.
+ *
+ * This is a weak no-op function in Open Enclave. The enclave application
+ * may override this function with a strong version. The strong version
+ * may add memory regions using the following functions.
+ *
+ *     - oe_region_start()
+ *     - oe_region_add_page()
+ *     - oe_region_end()
+ *
+ * For each region, the function should call **oe_region_start()**, followed
+ * by one more invocations of **oe_region_add_page()** and then finalized by
+ * calling **oe_region_end()**.
+ *
+ * @param[in] context the context parameter from **oe_add_regions()**
+ * @param[in] vaddr the beginning virtual address of the region
+ * @param[in] elf_image true if the region contains an ELF image
```

Comment:
```suggestion
 * @param[in] context The context parameter from **oe_add_regions()**
 * @param[in] vaddr The beginning virtual address of the region
 * @param[in] elf_image True if the region contains an ELF image
```
OE style is to capitalize the description